### DISCUSSION TITLE
Move Learn/JS frameworks guides under their own parent

### DISCUSSION THREAD

**Discussion by wbamberg** [mdn/mdn-community#310] [Content]
In the Learn docs, we have [Tools and testing](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing), and then under that [Tools and testing/Understanding client-side JavaScript frameworks](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks). And then under that we have a collection of guide pages, that fall into six quite well defined categories:

- general guides about frameworks (e.g. [Tools and testing/Understanding client-side JavaScript frameworks/Introduction to client-side frameworks](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/Introduction))
- a collection of tutorials for each of the frameworks covered: React, Ember, Svelte, Vue, Angular.

In the sidebar, these six categories are presented as six distinct groups:

<img width="294" alt="Screen Shot 2022-12-22 at 10 54 39 AM" src="https://user-images.githubusercontent.com/432915/209206674-0e269f71-42e1-4fee-b356-5594d4e58f99.png">

...but on disk/URL hierarchy, they are not - [all the pages live directly under Tools and testing/Understanding client-side JavaScript frameworks](https://github.com/mdn/content/tree/main/files/en-us/learn/tools_and_testing/client-side_javascript_frameworks).

I'd like to suggest we reorg them a little. Specifically:
- move "Client-side JavaScript frameworks" out from under "Tools and testing" and into the top level
- create pages for each framework under there, and have that framework's pages under those new pages

Like this:

```
- Learn
        - Tools and testing
                - Client-side web development tools
                        - Command line crash course
                        - etc
                - GitHub
                        - etc
                - Cross-browser testing
                        - etc
        - Client-side JavaScript Frameworks
                - Overview
                        - Framework main features
                - React
                        - Getting started with React
                        - etc
                - Ember
                        - Getting started with Ember
                        - etc
                - Vue
                - etc
```

My immediate motivation for this is that I think we could greatly simplify the Learn sidebar macro if the sidebar categories map to directories. But I also think this organization makes sense.

I've not been very closely involved with this end of Learn though, so I'd like to know if anyone thinks this is a Bad Idea.

**Comment by Josh-Cena** [comment]
Every time I look at this part of MDN I feel sad inside. The tutorial is nowhere near the official docs in terms of breadth and depth, and it always gets helplessly outdated because no-one updates them except new learners trying to help each other. Is it really relevant in the long term?

**Reply by wbamberg** [reply]
Are you talking specifically about the JS frameworks stuff? I agree about that and think we should never have written these docs in the first place. But I don't want my relatively small suggestion to get bogged down in a bigger topic, so can we keep this discussion focused and have a separate thread for the other thing.

**Reply by Josh-Cena** [reply]
Indeed, the client-side frameworks part. I feel like it's better if we can freeze it as much as possible without touching the structure—but if you feel it's impossible to avoid for the sidebars work, then I'm okay with anything being done with this section, with the caveat that the section needs to be entirely reconsidered in the near future.

**Comment by hamishwillee** [comment]
Yes, we should reorganise as you have suggested.

I agree that there are bigger questions around the lifespan of this whole section, but even that is conceptually easier to handle if all the content we're talking about is grouped.

I can do this if you want?

**Comment by dipikabh** [comment]
Thanks Will for raising this, and thanks everyone for sharing your thoughts.

Since the time this discussion was opened, the Learn area has undergone significant changes, and the context of the original proposal has shifted. So I'll be closing this discussion.
Please feel free to open a new one if you see any outstanding areas for improvement based on this section's current state on MDN.

Related to the framework content, we're working toward publishing a sunsetting policy for content, which will help us better handle that content going forward.

**Reply by wbamberg** [reply]
It seems like none of the frameworks other than React appear in the Learn sidebar at all any more, which doesn't seem like an improvement. 

<img width="232" height="460" alt="Screenshot 2026-02-04 at 4 38 04 PM" src="https://github.com/user-attachments/assets/473b2566-f6fa-44c5-a894-f9940b6740bc" />

Also, side issue, it looks the sidebar no longer scrolls so the current page is visible in it? Is that intentional?
